dotnet/generateShortUrl/src/Index.cs

```diff
@@ -0,0 +1,140 @@
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+public async Task<RuntimeResponse> Main(RuntimeRequest req, RuntimeResponse res)
+{
+  try
+  {
+    // Check if payload is present
+
+    if(String.IsNullOrEmpty(req.Payload))
+    {
+      return res.Json(new()
+      {
+        { "success", false },
+        { "message", "Payload is missing" }
+      });
+    }
+
+    var payload = JsonConvert.DeserializeObject<Dictionary<string, string>>(req.Payload);
+
+    // Check if provider and URL to be shortened are present in the payload
+
+    if(!(payload.ContainsKey("provider") && payload.ContainsKey("url")))
+    {
+      return res.Json(new()
+      {
+        { "success", false },
+        { "message", "Payload must contain a provider and a URL" }
+      });
+    }
+
+    string provider = payload["provider"];
+    string url = payload["url"];
+
+    string apiKey = "";
+    string uri = "";
+    Dictionary<string, string> body = new Dictionary<string, string>();
+    string json = "";
+
+    // Preparing API info based on chosen provider
+
+    if(payload["provider"].Equals("bitly"))
+    {
+      apiKey = req.Variables["BITLY_API_KEY"];
```

Comment:
Let's add check that ensures API key is provided.

Reply:
Same for tiny URL

Reply:
Made the changes